nntrainer/layers/embedding.cpp

```diff
@@ -64,7 +90,31 @@ void EmbeddingLayer::setProperty(const PropertyType type,
 }
 
 void EmbeddingLayer::forwarding(bool training) {
-  // NYI
+  Tensor &weight =
+    weightAt(static_cast<int>(EmbeddingParams::weight)).getVariableRef();
+  Tensor &hidden_ = net_hidden[0]->getVariableRef();
+  Tensor &input_ = net_input[0]->getVariableRef();
+
+  for (unsigned int b = 0; b < input_.getDim().batch(); ++b) {
+    Tensor in_sub = input_.getBatchSlice(b, 1);
+    Tensor out_sub = hidden_.getBatchSlice(b, 1);
+    float *in_data = in_sub.getData();
+
+    for (unsigned int i = 0; i < in_length; ++i) {
+      if (in_data[i] < 0)
+        continue;
+
+      float *weight_data =
+        &(weight.getData()[static_cast<uint>(in_data[i]) * out_dim]);
+      float *out_data = &(out_sub.getData()[i * out_dim]);
+      for (unsigned int j = 0; j < out_dim; ++j) {
```

Comment:
use `memcpy` or `scopy` instead of this loop?

Reply:
already done in #988